extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLOverWebSocketHandler.java

```diff
@@ -54,9 +58,31 @@ protected void doHandle(final RoutingContext ctx) {
                             serverWebSocket.close();
                             return;
                     }
+
+                    QuarkusHttpUser user = (QuarkusHttpUser) ctx.user();
+                    ScheduledFuture<?> authExpiryFuture = null;
+                    if (user != null) {
+                        //close the connection when the identity expires
+                        Long expire = user.getSecurityIdentity().getAttribute("quarkus.identity.expire-time");
+                        if (expire != null) {
+                            authExpiryFuture = ((ConnectionBase) ctx.request().connection()).channel().eventLoop()
+                                    .schedule(() -> {
```

Comment:
Do you have access to a Vert.x instance? It would be easier to just use vertx.setTimer(expire * 1000 - System.currentTimeMillis(), x -> { ... }).

Another reason to do that is that you are using the netty API directly which we want to avoid (due to the future netty 4 -> netty 5 migration)

Reply:
Cancellation would be handled by storing the timerId returned by setTimer. 
Note that it's simpler to use the bare Vert.x API in this case.

TIP: long cancellation = -1, and not 0 by default as the first one will be 0.